### ISSUE TITLE
Taxonomy in media list

### ISSUE THREAD

**Issue by muratmaga** [MorphoSource/morphosource-api#3]
@JulieWinchester we are rebuilding the MorphoSourceImport module with the new API and the python package @johnbradley put together (https://github.com/Imageomics/pyMorphoSource)

However, we need to be able to filter list of media by higher taxonomy (e.g., Primates, Mammalia etc), otherwise it is very hard to know what to search for (taxonomy at media seems limited to binomial species names only). We discussed a few solution [here](https://github.com/Imageomics/pyMorphoSource/issues/8#issuecomment-1791445108). I wonder if you have suggestions, or possible to implement this inthe API?

**Comment by JulieWinchester** [comment]
Hi @muratmaga and @johnbradley! Just to restate my impression of the request(s) from the other thread, they are:

* That it would be good if there was a way from the media search endpoint to facet or query media by a higher taxonomy term, the same way it's possible to do so for physical object search? Like in this physical object page search that facets by GBIF taxonomy terms to only list objects corresponding to Chordata: https://www.morphosource.org/catalog/objects?f%5Bexternal_taxonomy_ssim%5D%5B%5D=Chordata&locale=en&search_field=all_fields
* That it would be good if physical object API endpoints returned more information relating to GBIF taxonomy, including higher taxonomy?

I imagine that we can probably do something on the API level to help with this stuff, as well as hopefully the other issues that have been opened recently. We're grateful that you all are putting the API through testing, and letting us know the paint points that exist! Incidentally, it is intentional that the per_page parameter maxes out at 1000, though you can use the page parameter to page through results. We could be doing a much better job of documenting possible parameters that can be used with the API, and we'll try to improve that.

**Comment by johnbradley** [comment]
Thanks @JulieWinchester.

**Comment by JulieWinchester** [comment]
Hi @johnbradley and @muratmaga , I've pushed out a temporary draft of version 0.5.0 of the API, which reflects changes we've made on our dev server but have not yet pushed to production: https://morphosource.stoplight.io/docs/morphosource-api/rm6bqdolcidct-morpho-source-rest-api

If you want to take a look at this, especially the Search Media Records page (https://morphosource.stoplight.io/docs/morphosource-api/2d534286cca2e-search-media-records), let me know if it looks like this will address the issue described here?

Some of the changes we've made in API 0.5.0:
* Added field Taxonomy (GBIF) AKA `taxonomy_gbif` to Media and Physical Object search endpoints as a filter facet field and to Media and Physical Object responses as a response data field. This is an array of individual taxonomy terms ("Animalia", "Chordata", etc.) associated with a Media or Physical Object via a GBIF-aggregated taxonomy record. Around half of the specimens in MorphoSource currently have GBIF taxonomy linkages.
* Changed field for Media and Physical Objects from `taxonomy` to `taxonomy_name` to show the distinction from `taxonomy_gbif`. This is for binomial names or for taxonomies without genus or species information, it will be a higher order name combined with "indet."
* Simplified filter facet URL query parameter structure to a simpler `f.field_name` format. Also simplified filter facet field names significantly. 
* Improved documentation significantly. The Media and Physical Object response schemas now have field definitions and examples. For single-record query endpoints and multiple-record search endpoints, we've added details on query parameters and specifically filter facet fields, with example values in some cases.
